### ISSUE TITLE
Segfaults on exit

### ISSUE THREAD

**Issue by timsueberkrueb** [lirios/browser#62]
This might also be the reason for cookies not getting saved on exit.

`gdb` backtrace:
```
#0  0x00000000000403e8 in ?? ()
#1  0x00007fffeb42074e in QQuickWindow::mouseGrabberItem() const ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#2  0x00007fffeb422c68 in QQuickWindowPrivate::removeGrabber(QQuickItem*, bool, bool) () from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#3  0x00007fffeb411e3e in QQuickItemPrivate::derefWindow() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#4  0x00007fffeb412475 in QQuickItem::setParentItem(QQuickItem*) ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#5  0x00007fffeb412dbc in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#6  0x00007fffbfbe9fd0 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Layouts/libqquicklayoutsplugin.so
#7  0x00007fffbfbe7084 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Layouts/libqquicklayoutsplugin.so
#8  0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#9  0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#10 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#11 0x00007fffeb42ddbe in ?? ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
---Type <return> to continue, or q <return> to quit---
#12 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#13 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#14 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#15 0x00007fffeb42dd06 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#16 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#17 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#18 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#19 0x00007fffeb42e286 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#20 0x00007ffff6be5d83 in QQmlDelegateModel::~QQmlDelegateModel() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Qml.so.5
#21 0x00007ffff6be5e69 in QQmlDelegateModel::~QQmlDelegateModel() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Qml.so.5
#22 0x00007fffeb4e1ffa in QQuickItemView::~QQuickItemView() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#23 0x00007fffeb42e0d6 in ?? ()
---Type <return> to continue, or q <return> to quit---
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#24 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#25 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#26 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#27 0x00007fffbfbe9fd0 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Layouts/libqquicklayoutsplugin.so
#28 0x00007fffbfbe7084 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Layouts/libqquicklayoutsplugin.so
#29 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#30 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#31 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#32 0x00007fffeb42dd06 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#33 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#34 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
---Type <return> to continue, or q <return> to quit---
#35 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#36 0x00007fffbfbe9fd0 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Layouts/libqquicklayoutsplugin.so
#37 0x00007fffbfbe7114 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Layouts/libqquicklayoutsplugin.so
#38 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#39 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#40 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#41 0x00007fffeb42dd06 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#42 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#43 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#44 0x00007fffeb412ec6 in QQuickItem::~QQuickItem() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#45 0x00007fffbec9f45e in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Templates.2/libqtquicktemplates2plugin.so
---Type <return> to continue, or q <return> to quit---
#46 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#47 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#48 0x00007fffeb41fea4 in QQuickWindow::~QQuickWindow() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Quick.so.5
#49 0x00007fffbec9a411 in ?? ()
   from /home/tim/Qt/5.9/gcc_64/qml/QtQuick/Templates.2/libqtquicktemplates2plugin.so
#50 0x00007ffff7938b83 in QObjectPrivate::deleteChildren() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#51 0x00007ffff7942914 in QObject::~QObject() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Core.so.5
#52 0x00007ffff6b3185b in ?? () from /home/tim/Qt/5.9/gcc_64/lib/libQt5Qml.so.5
#53 0x00007ffff6bb3f4e in QQmlApplicationEnginePrivate::cleanUp() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Qml.so.5
#54 0x00007ffff6bb3fd8 in QQmlApplicationEngine::~QQmlApplicationEngine() ()
   from /home/tim/Qt/5.9/gcc_64/lib/libQt5Qml.so.5
#55 0x000055555556c54a in main (argc=1, argv=0x7fffffffda88)
    at /home/tim/Dev/Liri/browser/src/main/main.cpp:100
```

**Comment by timsueberkrueb** [comment]
Looks like this upstream Qt bug: https://bugreports.qt.io/browse/QTBUG-61434.

**Comment by timsueberkrueb** [comment]
Only occurs with Qt 5.9/5.9.1.

**Comment by timsueberkrueb** [comment]
Yay, should be [fixed](https://bugreports.qt.io/browse/QTBUG-61434) in Qt 5.9.3.

**Comment by timsueberkrueb** [comment]
Fixed since Qt 5.9.3.
